### PR TITLE
Add more UTs for SonarAnalysisConfig

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Related to #4156

This is draft, rebase will be needed when #4179 is merged. First squash commit doesn't need a review.

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/SonarAnalysisContextTest.cs [hunk 1]
```diff
@@ -312,6 +316,40 @@ public void ProjectConfiguration_WhenInvalidXml_ThrowException()
...
+        [DataTestMethod]
+        [DataRow(true, false, MainTag)]
+        [DataRow(true, false, MainTag, TestTag)]
+        [DataRow(true, true, TestTag)]
+        [DataRow(true, true, MainTag, TestTag)]
+        [DataRow(false, false, TestTag)]
+        [DataRow(false, false, TestTag, TestTag)]
+        [DataRow(false, true, MainTag)]
+        [DataRow(false, true, MainTag, MainTag)]
+        public void IsAnalysisScopeMatching_SingleDiagnostis_WithOneOrMoreScopes(bool expectedResult, bool isTestProject, params string[] singleDiagnosticTags)
```
Comment (andrei-epure-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Helpers/SonarAnalysisContextTest.cs:332]: I'd put expectedResult at the end, for consistency (input, output) like in `public void IsTestProject_Standalone(ProjectType projectType, bool expectedResult)`

Comment (andrei-epure-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/Helpers/SonarAnalysisContextTest.cs:332]: and rename `singleDiagnosticTags` to `testRuleScope`

Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/Helpers/SonarAnalysisContextTest.cs:332]: I have it intentionally at the beginning to have table-like structure (easier to read), as the argument count varies. And I couldn't use `params` argument like that (more syntax noise with worse readability).

**Review by andrei-epure-sonarsource — APPROVED** [review]
great tests!

**Pull request merged**